autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -185,31 +186,41 @@ def update_execution(status: ExecutionStatus) -> ExecutionResult:
             ):
                 yield execution
 
-        # Release lock on credentials ASAP
```

Comment:
moving this piece to the finally block

Reply:
i think the goal was to release ASAP here so we'd want that put back as it will block any other block from executing

Reply:
There is no code between this line and finally block where the cred_lock release is done.